src/tensors/gpu/integer_tools.cu

```diff
@@ -600,6 +794,35 @@ namespace integer {
         CUDA_CHECK(cudaGetLastError()); // Get errors from kernel launches
     }
 
+    __global__ void quantizeFP16(const half *input,
+                             int8_t *output,
+                             size_t items,
+                             const float *quantMultAddr) {
+      const float quantMult
+          = *quantMultAddr;  //@TODO ask nvidia if this is the most efficient way to do this here
+      size_t x = blockIdx.x * blockDim.x + threadIdx.x;
+      int i = threadIdx.x;
+      __shared__ float share[256];  // Not sure if shared memory is necessary here to take advnatage
+                                    // of globale memory burst
+      if(x < items) {
+        share[i] = __half2float(input[x]);
```

Comment:
I think there's a direct round function that operates on halfs, I will try to fix this myself though